openstudiocore/resources/model/OpenStudio.idd

```diff
@@ -1980,6 +1980,39 @@ OS:BuildingStory,
        \units m
        \minimum> 0
 
+OS:BuildingUnit,
+       \min-fields 4
+       \extensible:3
+  A1,  \field Handle
+       \type handle
+       \required-field
+  A2,  \field Name
+       \type alpha
+       \reference BuildingUnitNames
+       \required-field
+  A3,  \field Rendering Color
+       \type object-list
+       \object-list GroupRenderingNames
+  A4,  \field Building Unit Type
+       \type choice
+       \key Residential
+       \key NonResidential
+       \default Residential
+  A5,  \field Building Unit Feature Name
+       \type alpha
+       \begin-extensible
+       \required-field
+  A6,  \field Building Unit Feature Data Type
+       \type choice
+       \key alpha
```

Comment:
Not yet, that Variant class is only on OpenStudio 2.X branch for now.  Eventually I think your class should have a method:

    boost::optional<openstudio::Variant> getFeature(const std::string& name) const;